contributors/design-proposals/cluster-info-configmap.md

```diff
@@ -0,0 +1,25 @@
+# Persist cluster-wide data in ConfigMap
+
+## Abstract
+
+We want to allow cluster-wide data to be persisted for general purposes, currently for us (service catalog) is to focus on the ability to identify different clusters that access our service broker with a 'cluster-id'.
+
+## Motivation
+
+Currently there is no way for service catalog api server to identify the different clusters that access our service brokers. The goal of this design is to have a unify storage for cluster related information. The information should be retained and survive between component failures. 
+
+## Detailed design
+
+A configmap `cluster-info` is created in the `default` namespace at the starup of the controller. The initial content of the map will be as the following:
+
+```
+apiVersion: v1
+data:
+  cluster-id: uuid-of-the-cluster
```

Comment:
until now, things that needed to identify a cluster among many (federation is the only one that comes to mind) had their own id-to-cluster-address mapping. this makes a cluster aware of its own id, which is a change. some questions that come to mind:
* is the configmap value or the flag value fed to the controller that creates the configmap authoritative?
* how do multiple controllers coordinate this value? what if they disagree with each other? with an existing value in the configmap?
* what are the implications if the configmap value changes or is removed?
* is the uuid intended to be randomly generated? assigned? by whom?
* what are the implications if two clusters claim to have the same id?
* where is this id denormalized to?
* what is an appropriate use of this id?